### ISSUE TITLE
auto-form - custom component example

### ISSUE THREAD

**Issue by gashcrumb** [syndesisio/syndesis-react#353]
## This is a...

<pre><code>
[x] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Documentation issue or request
</code></pre>

## Description

I'd like to have some kind of custom component example for auto-form that doesn't actually live in the auto-form package.  Mostly as advertising that it's a thing that can be done :-)  

I think ideally a custom auto-form component would live in the `syndesis` package, *however* I did find when I started to create one that the problem is we don't want to have pf3 `FormGroup` and other related components being imported into `syndesis`.  So maybe the `ui` package is a better place to keep custom form components?  That'd mean making `auto-form` a dependency of the `ui`, unless I guess if I maybe extract out the `auto-form` interfaces and types into a separate package.

@riccardo-forina thoughts?

**Comment by gashcrumb** [comment]
Basically a custom form component is just UI component that uses the `field.onChange` and `field.onBlur` props passed in by Formik to update the property value when appropriate.  Technically I'd call the "duration" control an example of a custom component really but I already baked it into auto-form.

**Comment by riccardo-forina** [comment]
maybe a @syndesis/autoform-widgets? It could depend on PF directly, bypassing the UI (which should contain stuff app specific anyway).

**Comment by gashcrumb** [comment]
Yeah maybe like `@syndesis/autoform-pf3`, `@syndesis/autoform-pf4` I guess perhaps would be handy at migration time certainly.

My other thought was perhaps [this map](https://github.com/syndesisio/syndesis-react/blob/master/app/ui-react/packages/auto-form/src/FormBuilder.tsx#L43-L50) could be passed in externally via a prop, then the app can basically just tell autoform exactly what widget to use.  Could even do it via overriding things in that `customComponents` prop.  

If we think #160 would be super-helpful we might wind up kinda having to do a similar thing as part of supporting that component, or just not make it part of auto-form and instead make it part of the app.
